esphome/components/midea/climate.py

```diff
@@ -0,0 +1,284 @@
+    cg.add(var.set_temperature(template_))
+
+
+# Toggle Display action
+@register_action(
+    "display_toggle",
+    DisplayToggleAction,
+    cv.Schema({}),
+)
+async def display_toggle_to_code(var, config, args):
+    pass
+
+
+# Swing Step action
+@register_action(
+    "swing_step",
+    SwingStepAction,
+    cv.Schema({}),
+)
+async def swing_step_to_code(var, config, args):
+    pass
+
+
+# Beeper On action
+@register_action(
+    "beeper_on",
+    BeeperOnAction,
+    cv.Schema({}),
+)
+async def beeper_on_to_code(var, config, args):
+    pass
+
+
+# Beeper Off action
+@register_action(
+    "beeper_off",
+    BeeperOffAction,
+    cv.Schema({}),
+)
+async def beeper_off_to_code(var, config, args):
+    pass
+
+
+# Power On action
+@register_action(
+    "power_on",
+    PowerOnAction,
+    cv.Schema({}),
+)
+async def power_on_to_code(var, config, args):
+    pass
+
+
+# Power Off action
+@register_action(
+    "power_off",
+    PowerOffAction,
+    cv.Schema({}),
+)
+async def power_off_to_code(var, config, args):
+    pass
+
+
+async def to_code(config):
+    var = cg.new_Pvariable(config[CONF_ID])
+    await cg.register_component(var, config)
+    await uart.register_uart_device(var, config)
+    await climate.register_climate(var, config)
+    cg.add(var.set_period(config[CONF_PERIOD].total_milliseconds))
+    cg.add(var.set_response_timeout(config[CONF_TIMEOUT].total_milliseconds))
+    cg.add(var.set_request_attempts(config[CONF_NUM_ATTEMPTS]))
+    if CONF_TRANSMITTER_ID in config:
+        cg.add_define("USE_REMOTE_TRANSMITTER")
+        transmitter_ = await cg.get_variable(config[CONF_TRANSMITTER_ID])
+        cg.add(var.set_transmitter(transmitter_))
+    cg.add(var.set_beeper_feedback(config[CONF_BEEPER]))
+    cg.add(var.set_autoconf(config[CONF_AUTOCONF]))
+    if CONF_SUPPORTED_MODES in config:
+        cg.add(var.set_supported_modes(config[CONF_SUPPORTED_MODES]))
+    if CONF_SUPPORTED_SWING_MODES in config:
+        cg.add(var.set_supported_swing_modes(config[CONF_SUPPORTED_SWING_MODES]))
+    if CONF_SUPPORTED_PRESETS in config:
+        cg.add(var.set_supported_presets(config[CONF_SUPPORTED_PRESETS]))
+    if CONF_CUSTOM_PRESETS in config:
+        cg.add(var.set_custom_presets(config[CONF_CUSTOM_PRESETS]))
+    if CONF_CUSTOM_FAN_MODES in config:
+        cg.add(var.set_custom_fan_modes(config[CONF_CUSTOM_FAN_MODES]))
+    if CONF_OUTDOOR_TEMPERATURE in config:
+        sens = await sensor.new_sensor(config[CONF_OUTDOOR_TEMPERATURE])
+        cg.add(var.set_outdoor_temperature_sensor(sens))
+    if CONF_POWER_USAGE in config:
+        sens = await sensor.new_sensor(config[CONF_POWER_USAGE])
+        cg.add(var.set_power_sensor(sens))
+    if CONF_HUMIDITY_SETPOINT in config:
+        sens = await sensor.new_sensor(config[CONF_HUMIDITY_SETPOINT])
+        cg.add(var.set_humidity_setpoint_sensor(sens))
+    cg.add_library("dudanov/MideaUART", "^1.1.0")
```

Comment:
Would prefer if this was not a caret target
```suggestion
    cg.add_library("dudanov/MideaUART", "1.1.0")
```

Reply:
Done